### DISCUSSION TITLE
MySQL upsert

### DISCUSSION THREAD

**Discussion by jossefaz** [sqlalchemy/sqlalchemy#8069] [Usage Questions] [answered]
Is there any equivalent for MySQL today for the ON_CONFLICT upsert (from PostgreSQL dialect), like explained here :
https://docs.sqlalchemy.org/en/14/orm/persistence_techniques.html#using-postgresql-on-conflict-with-returning-to-return-upserted-orm-objects ?

**Comment by zzzeek** [comment, marked as answer]
MySQL's "on conflict" equivalent is ON DUPLICATE KEY UPDATE, which we support here: https://docs.sqlalchemy.org/en/14/dialects/mysql.html#insert-on-duplicate-key-update-upsert

however, MySQL's construct does not support "RETURNING" of matched rows; MariaDB's does, however SQLAlchemy has yet to add support for MariaDB's RETURNING, this will be in release 2.0 of SQLAlchemy.
